src/playback-watcher.js

```diff
@@ -123,6 +154,10 @@ export default class PlaybackWatcher {
     let seekable = this.seekable();
     let currentTime = this.tech_.currentTime();
 
+    if (this.tech_.seeking() && this.fixBadSeeks_()) {
```

Comment:
The waiting event can fire and cause `waiting_` to run independent of `checkCurrentTime_`